vllm/attention/backends/xformers.py

```diff
@@ -116,65 +145,196 @@ def __post_init__(self):
         # will not appear in the __repr__ and __init__
         self.attn_bias: Optional[List[AttentionBias]] = None
 
+    @property
+    def has_valid_cross_attn_metadata(self):
+        # No cross-attention metadata is present whatsoever
```

Comment:
can we return (self.cross_seq_lens is not
                 None) and (self.cross_slot_mapping is not
                            None) and (self.cross_block_tables is not None) here and in the do_cross_attn setter function assert it?

Reply:
Just pushed a fix for this.